src/utils.cpp

```diff
@@ -1947,17 +1947,26 @@ void magic_loss(struct char_data *ch, int magic, bool msg)
 }
 
 // Return true if the character has a kit of the given type, false otherwise.
-#define IS_KIT(obj, type) (GET_OBJ_TYPE((obj)) == ITEM_WORKSHOP && GET_WORKSHOP_TYPE((obj)) == type && GET_WORKSHOP_GRADE((obj)) == TYPE_KIT)
+// Note: some kits may still have workshop grade of 0 instead of TYPE_KIT.
+#define IS_KIT(obj, type) ( GET_OBJ_TYPE((obj)) == ITEM_WORKSHOP && GET_WORKSHOP_TYPE((obj)) == type && (GET_WORKSHOP_GRADE((obj)) == TYPE_KIT || GET_WORKSHOP_GRADE((obj)) == 0) )
 bool has_kit(struct char_data * ch, int type)
 {
-  for (struct obj_data *o = ch->carrying; o; o = o->next_content)
-    if (IS_KIT(o, type))
-      return TRUE;
+  struct obj_data *obj;
+  bool found = FALSE;
+  
+  for (obj = ch->carrying; obj && !found; obj = obj->next_content) {
+    if (IS_KIT(obj, type)) {
+      found = TRUE;
```

Comment:
Is there a reason behind the change from `return TRUE` to `found = TRUE`? Short-circuiting execution with a return the moment you've established its value means the server spends less time looping over things unnecessarily.